src/main/java/org/terasology/launcher/config/PackageAdapter.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2019 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.terasology.launcher.config;
+
+import com.google.gson.TypeAdapter;
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonWriter;
+import org.terasology.launcher.packages.Package;
+
+import java.io.IOException;
+
+class PackageAdapter extends TypeAdapter<Package> {
```

Comment:
Why is the package URL not serialized by this type adapter?

Reply:
This part is actually related to #463 and #464. 
I was thinking of redesigning parts of Package Manager API to support package info (changelog, etc.) directly from the `PackageDatabase`. 
In a nutshell, that'll end up changing the database into a `Map<Package, PackageInfo>` from the current `List<Package>`... The `PackageInfo` can then contain its `url`, `status`, `changelog`, etc.
Also, I don't think serializing the URL into JSON there would be a good idea: we'll need to detect tampering later.